main.tf

```diff
@@ -203,11 +203,11 @@ resource "aws_route" "database_internet_gateway" {
 }
 
 resource "aws_route" "database_nat_gateway" {
-  count = var.create_vpc && var.create_database_subnet_route_table && length(var.database_subnets) > 0 && false == var.create_database_internet_gateway_route && var.create_database_nat_gateway_route && var.enable_nat_gateway ? local.nat_gateway_count : 0
+  count = var.create_vpc && var.create_database_subnet_route_table && length(var.database_subnets) > 0 && false == var.create_database_internet_gateway_route && var.create_database_nat_gateway_route && var.enable_nat_gateway ? 1 : 0
 
-  route_table_id         = element(aws_route_table.private.*.id, count.index)
+  route_table_id         = aws_route_table.database[0].id
```

Comment:
I think here we need to use `coalesce` to use the correct value of `aws_route_table` to use for this route (it can be either `database` or `private`, depending on configuration).

Also, leave `local.nat_gateway_count` as it was because there can be several NAT gateways.

Reply:
What's the best way for me to contribute here? Should I push changes to @KyleMartin901 branch? or continue the PR I created (#476)?

Reply:
I don't see the possibility of needing a route for a database dedicated NAT Gateway on a private subnet route table. @antonbabenko Can you elaborate on why the private subnets would be needed here?

Reply:
@antonbabenko Can we continue this change in my pull request #476?